src/WinRT.Runtime/Projections/Nullable.cs

```diff
@@ -2203,6 +2006,139 @@ internal static class Nullable_Delegates
         public unsafe delegate int GetValueDelegateAbiDateTimeOffset(void* ptr, DateTimeOffset* result);
     }
 
+    internal static class NullableBlittable<T> where T: unmanaged
+    {
+        private readonly static Guid IID = NullableType.GetIIDForBuiltInType(typeof(T));
+
+        public static unsafe object GetValue(IInspectable inspectable)
+        {
+            IntPtr nullablePtr = IntPtr.Zero;
+            try
+            {
+                T __retval = default;
+                ExceptionHelpers.ThrowExceptionForHR(Marshal.QueryInterface(inspectable.ThisPtr, ref Unsafe.AsRef(in IID), out nullablePtr));
+                ExceptionHelpers.ThrowExceptionForHR((*(delegate* unmanaged[Stdcall]<IntPtr, void*, int>**)nullablePtr)[6](nullablePtr, &__retval));
+                return __retval;
+            }
+            finally
+            {
+                Marshal.Release(nullablePtr);
+            }
+        }
+    }
+
+    internal static class NullableType
+    {
+        internal static Guid GetIIDForBuiltInType(global::System.Type type)

```

Comment:
All these type checks where in a generic context before so that everything could be correctly elided, due to generic specialization over value types. If we're just branching on a `Type` instance now, we're losing that, meaning we'll end up rooting every single path. This is not so much a big deal here but it is a bit below where each path also triggers a `NullableBlittable<T>` instantiation and delegate creation. Can we keep these methods generic?

Reply:
I can easily make `GetIIDForBuiltInType` generic.  But `GetValueFactory` is a bit harder given type is a parameter and it is called from `CreateTypedRcwFactory`.

Reply:
Ah, I see, that's all wired up from the other RCW factory path. Alright let me run some size diffs first, maybe it doesn't hurt too much and we can just say the small size regression there is worth it to make this work with the current approach. In theory that type is very small so maybe we're lucky and it's not a big deal. Be right back 😄

Reply:
Just updating here too for context, I guess we can just look at the two things separately. For the one creating RCW factories below, we might just have to leave that one as is at least for now. For this one, we can definitely make it generic though, which can help with inlining and also dropping some fallback code paths. In fact, what I'm thinking is we could change this method to do everything for arbitrary types, like so:

```csharp
public static Guid GetIID<T>()
{
    if (typeof(T) == typeof(int)) return Nullable_int.IID;
    if (typeof(T) == typeof(byte)) return Nullable_byte.IID;
    if (typeof(T) == typeof(bool)) return Nullable_bool.IID;
    if (typeof(T) == typeof(sbyte)) return Nullable_sbyte.IID;
    if (typeof(T) == typeof(short)) return Nullable_short.IID;
    if (typeof(T) == typeof(ushort)) return Nullable_ushort.IID;
    if (typeof(T) == typeof(char)) return Nullable_char.IID;
    if (typeof(T) == typeof(uint)) return Nullable_uint.IID;
    if (typeof(T) == typeof(long)) return Nullable_long.IID;
    if (typeof(T) == typeof(ulong)) return Nullable_ulong.IID;
    if (typeof(T) == typeof(float)) return Nullable_float.IID;
    if (typeof(T) == typeof(double)) return Nullable_double.IID;
    if (typeof(T) == typeof(Guid)) return Nullable_guid.IID;
    if (typeof(T) == typeof(global::System.Type)) return Nullable_Type.IID;
    if (typeof(T) == typeof(global::System.TimeSpan)) return Nullable_TimeSpan.IID;
    if (typeof(T) == typeof(global::System.DateTimeOffset)) return Nullable_DateTimeOffset.IID;
    if (typeof(T) == typeof(global::Windows.Foundation.Point)) return IReferenceIIDs.IReferenceOfPoint_IID;
    if (typeof(T) == typeof(global::Windows.Foundation.Size)) return IReferenceIIDs.IReferenceOfSize_IID;
    if (typeof(T) == typeof(global::Windows.Foundation.Rect)) return IReferenceIIDs.IReferenceOfRect_IID;
    if (typeof(T) == typeof(global::System.Numerics.Matrix3x2)) return IReferenceIIDs.IReferenceMatrix3x2_IID;
    if (typeof(T) == typeof(global::System.Numerics.Matrix4x4)) return IReferenceIIDs.IReferenceMatrix4x4_IID;
    if (typeof(T) == typeof(global::System.Numerics.Plane)) return IReferenceIIDs.IReferencePlane_IID;
    if (typeof(T) == typeof(global::System.Numerics.Quaternion)) return IReferenceIIDs.IReferenceQuaternion_IID;
    if (typeof(T) == typeof(global::System.Numerics.Vector2)) return IReferenceIIDs.IReferenceVector2_IID;
    if (typeof(T) == typeof(global::System.Numerics.Vector3)) return IReferenceIIDs.IReferenceVector3_IID;
    if (typeof(T) == typeof(global::System.Numerics.Vector4)) return IReferenceIIDs.IReferenceVector4_IID;

    return GuidGenerator.CreateIID(typeof(Nullable<T>));
}
```
That `PIID` field initializer above can then also just be `NullableType.GetIID<T>();`.
(the method can also be public, since it's in an internal type already).

Reply:
Addressed.